test/automation/src/sql/notebook.ts

```diff
@@ -95,4 +103,20 @@ export class NotebookToolbar {
 		const kernelDropdownValue = `${NotebookToolbar.toolbarSelector} select[id="kernel-dropdown"][title="${kernel}"]`;
 		await this.code.waitForElement(kernelDropdownValue, undefined, 3000); // wait up to 5 minutes for kernel change
 	}
+
+	async trustNotebook(): Promise<void> {
+		const trustedButton = `${NotebookToolbar.toolbarSelector} a[class="action-label codicon notebook-button masked-icon icon-shield"]`;
+		const notTrustedButton = `${NotebookToolbar.toolbarSelector} a[class="action-label codicon notebook-button masked-icon icon-shield-x"]`;
+
+		await this.code.waitAndClick(NotebookToolbar.toolbarSelector);
+		if (this.code.waitForElement(notTrustedButton)) {
+			this.code.waitAndClick(notTrustedButton);
+			this.code.waitForElement(trustedButton);
+		}
+	}
+
+	async isTrusted(): Promise<void> {
+		const trustedButton = `${NotebookToolbar.toolbarSelector} a[class="action-label codicon notebook-button masked-icon icon-shield"]`;
+		await this.code.waitForElement(trustedButton);
```

Comment:
If it's not trusted isn't this going to time out? Seems better to me to just have the selector get the button without specifying the icon and then looking at the classes to determine which one it is.

Reply:
Changed this method name to waitForTrustedIcon to be more representative of what the method is doing.